### ISSUE TITLE
Rename conversion functions rescale_to_*

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#1234]
Many users have been confused by our image conversion functions before. Renaming them to rescale_to_{ubyte,uint,int,float} will more accurately reflect what they do. See #1229 for discussion.

**Comment by sciunto** [comment]
@scikit-image/core This is a ~ 5 years old proposal. Do we still agree that we want these rescale_to_* functions in scikit-image?

**Comment by jni** [comment]
@sciunto thanks for the ping! imho this should remain on our radar. It's just not a high priority because things are working fine...

**Comment by sciunto** [comment]
Thanks! I plan it for 1.0.

**Comment by alexandrejaguar** [comment]
Hey everyone,
I really like the idea, but thinking on the "beginner", I was wondering if `rescale_to...` wouldn't be confused with the `rescale` operation (ours on `skimage.transform.rescale`).
How about `convert_to_*`?

**Comment by hmaarrfk** [comment]
I seriously wonder what the root cause of the confusion is.

Most people have trouble understand why the scales are sometimes between 255 and sometimes between 0 and 1.

People generally are unformtable with image saving formats and scientific precision. Not sure if renaming stuff is the best way to reach that.

**Comment by alexandrejaguar** [comment]
@hmaarrfk I get your point, but being explicit in this case would be part of the solution. On the other hand, we could put an extended section on our docs explaining how to deal with image formats/precision; that could help solving the other issue you exposed.

**Comment by sciunto** [comment]
We have this: https://scikit-image.org/docs/stable/user_guide/data_types.html

**Comment by alexandrejaguar** [comment]
@sciunto yes, I remember this one. I was thinking on maybe extending it with some bells and whistles :)

**Comment by mkcor** [comment]
> Hey everyone,
> I really like the idea, but thinking on the "beginner", I was wondering if `rescale_to...` wouldn't be confused with the `rescale` operation (ours on `skimage.transform.rescale`).
> How about `convert_to_*`?

I changed my mind and prefer `rescale_`; I think that `convert_` refers to conversions between colour spaces.

**Comment by lagru** [comment]
We, @stefanv and @matthew-brett, had an extensive discussion about this yesterday. I'll attempt to summarize my understanding of it.

Renaming to the pattern `rescale_to_*` and other behavioral changes introduced in #7697 make sense from a close perspective. However, in the big picture they don't make sense when taking into account skimage2. With skimage2 we intend to no longer require rescaling for our inputs which makes these utility functions somewhat obsolete. Why put users through switching to `rescale_to_*` when we intend those function to be unnecessary in the close future. From that follows, that we leave those function until we remove them with skimage2.

After thinking about this more — since yesterday — I think slapping a deprecation warning on the `img_as` functions has dubious value. Before we remove the need for rescaling input in skimage2, users still need these functions. So a deprecation warning would merely warn them about an impeding change while the only course to silence this warning will be a warning filter. That's probably more annoying than helpful for our users.

With that I propose to close this issue without changing anything or merging #7697. Instead, we should consider removing these `img_as` functions with skimage2. What do you think?
